src/mbgl/map/tile_worker.cpp

```diff
@@ -161,101 +179,114 @@ void TileWorker::addBucketGeometries(Bucket& bucket, const GeometryTileLayer& la
+    applyLayoutProperty(PropertyKey::TextFont, styleBucket.layout, layout.text.font, z);
+    applyLayoutProperty(PropertyKey::TextMaxWidth, styleBucket.layout, layout.text.max_width, z);
+    applyLayoutProperty(PropertyKey::TextLineHeight, styleBucket.layout, layout.text.line_height, z);
+    applyLayoutProperty(PropertyKey::TextLetterSpacing, styleBucket.layout, layout.text.letter_spacing, z);
+    applyLayoutProperty(PropertyKey::TextMaxAngle, styleBucket.layout, layout.text.max_angle, z);
+    applyLayoutProperty(PropertyKey::TextRotate, styleBucket.layout, layout.text.rotate, z);
+    applyLayoutProperty(PropertyKey::TextPadding, styleBucket.layout, layout.text.padding, z);
+    applyLayoutProperty(PropertyKey::TextIgnorePlacement, styleBucket.layout, layout.text.ignore_placement, z);
+    applyLayoutProperty(PropertyKey::TextOptional, styleBucket.layout, layout.text.optional, z);
+    applyLayoutProperty(PropertyKey::TextJustify, styleBucket.layout, layout.text.justify, z);
+    applyLayoutProperty(PropertyKey::TextAnchor, styleBucket.layout, layout.text.anchor, z);
+    applyLayoutProperty(PropertyKey::TextKeepUpright, styleBucket.layout, layout.text.keep_upright, z);
+    applyLayoutProperty(PropertyKey::TextTransform, styleBucket.layout, layout.text.transform, z);
+    applyLayoutProperty(PropertyKey::TextOffset, styleBucket.layout, layout.text.offset, z);
+    applyLayoutProperty(PropertyKey::TextAllowOverlap, styleBucket.layout, layout.text.allow_overlap, z);
+
+    applyLayoutProperty(PropertyKey::IconSize, styleBucket.layout, layout.icon.size, z + 1);
+    applyLayoutProperty(PropertyKey::IconSize, styleBucket.layout, layout.icon.max_size, 18);
+    applyLayoutProperty(PropertyKey::TextSize, styleBucket.layout, layout.text.size, z + 1);
+    applyLayoutProperty(PropertyKey::TextSize, styleBucket.layout, layout.text.max_size, 18);
+
+    bucket->parseFeatures(layer, styleBucket.filter);
+
+    assert(style.glyphStore);
+    assert(style.sprite);
+    const bool needsDependencies = bucket->needsDependencies(*style.glyphStore, *style.sprite);
+    if (needsDependencies) {
+        // We cannot parse this bucket yet. Instead, we're saving it for later.
+        pending.emplace_back(styleBucket, std::move(bucket));
+    } else {
+        assert(style.spriteAtlas);
+        assert(style.glyphAtlas);
+        assert(style.glyphStore);
+        assert(collisionTile);
+        bucket->addFeatures(reinterpret_cast<uintptr_t>(this), *style.spriteAtlas,
+                            *style.glyphAtlas, *style.glyphStore, *collisionTile);
+        insertBucket(styleBucket.name, std::move(bucket));
     }
+}
 
-    // We do not proceed if the parser is in a "partial" state because
-    // the layer ordering needs to be respected when calculating text
-    // collisions. Although, at this point, we requested all the resources
-    // needed by this tile.
-    if (partialParse) {
```

Comment:
Does this change how partial parsing works? Previously, because `partialParse` was a member variable, if _any_ symbol layer needed dependencies, all subsequent symbol buckets would be treated as partial. Now it looks like only a bucket that `needsDependencies` is treated as partial. From my reading of the above comment, the previous behavior was intentional.